### PR TITLE
Enhancement: Use `WyriHaximus/github-action-composer-php-versions-in-range` to determine supported PHP version matrix

### PR DESCRIPTION
**Pull request by localheinz**
This pull request

- [x] uses [`WyriHaximus/github-action-composer-php-versions-in-range`](https://github.com/WyriHaximus/github-action-composer-php-versions-in-range) to determine the supported PHP version matrix

### CODE REVIEW

**Comment by sebastianbergmann** [comment]
If this works for us, then we should start with it in `8.5` and merge to `9.5` and `main` from there.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -306,10 +317,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        php-version:
-          - "7.2"
-          - "7.3"
-          - "7.4"
+        php-version: ${{ fromJson(needs.supported-versions-matrix.outputs.version) }}
```
Comment (localheinz) [inline .github/workflows/ci.yml:320]: @sebastianbergmann 

Looks like we will be running additional jobs here, for

- "8.0
- "8.1"
- "8.2"

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -361,10 +369,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        php-version:
-          - "8.0"
-          - "8.1"
-          - "8.2"
+        php-version: ${{ fromJson(needs.supported-versions-matrix.outputs.version) }}
```
Comment (localheinz) [inline .github/workflows/ci.yml:372]: @sebastianbergmann 

Looks like we will be running additional jobs here, for

- "7.2
- "7.3"
- "7.4"

**Comment by localheinz** [comment]
@sebastianbergmann 

Looks like this is blocked by https://github.com/WyriHaximus/github-action-composer-php-versions-in-range/pull/5.

**Comment by WyriHaximus** [comment]
@localheinz @sebastianbergmann Just tagged [`v1.5.0`](https://github.com/WyriHaximus/github-action-composer-php-versions-in-range/releases/tag/v1.5.0) on that action to unblock this PR. Since you're using `@v1` that new version will be used on the next run.

**Comment by WyriHaximus** [comment]
No, thank you @localheinz. Until a few weeks ago I never shared about this project, so having someone pick it up in their projects is exciting. Someone proposing it to a high profile project like this is very exciting. Thank you ❤️ .

**Comment by localheinz** [comment]
@sebastianbergmann suggested to take a look, @WyriHaximus, so that's why I opened a pull request!

🤓

**Comment by sebastianbergmann** [comment]
Thank you for your contribution. I appreciate the time you invested in preparing this pull request. However, I have decided not to merge it.

**Comment by WyriHaximus** [comment]
For my own insights, what made you decide not to merge this?

**Comment by sebastianbergmann** [comment]
On the `8.5` branch we have the jobs `test-phar` and `test-phar-without-code-coverage` that require different version ranges than the other jobs:

grafik

We could use your action for the other jobs, of course, but that would be inconsistent. That is why we decided against using it, at least for now.

**Comment by WyriHaximus** [comment]
Thank you for sharing. And I fully agree, that action was built with keeping things simple in mind.

**Pull request closed without merging**